src/Cli/dotnet/commands/dotnet-test/VSTestArgumentConverter.cs

```diff
@@ -1,58 +1,87 @@
-            ["--results-directory"] = "--resultsdirectory",
-            ["--arch"] = "--platform"
-        };
-
-        private readonly Dictionary<string, string> VerbosityMapping = new Dictionary<string, string>
-        {
-            ["q"] = "quiet",
-            ["m"] = "minimal",
-            ["n"] = "normal",
-            ["d"] = "detailed",
-            ["diag"] = "diagnostic"
-        };
-
-        private readonly string[] IgnoredArguments = new string[]
-        {
-            "-c",
-            "--configuration",
-            "--runtime",
-            "-o",
-            "--output",
-            "--no-build",
-            "--no-restore",
-            "--interactive",
-            "--testSessionId",
-            "--artifacts-processing-mode"
-        };
+        private static readonly ImmutableDictionary<string, string> s_argumentMapping
+            = new Dictionary<string, string>()
+            {
+                ["-h"] = "--help",
+                ["-s"] = "--settings",
+                ["-t"] = "--listtests",
+                // .NET 7 breaking change, before we had ["-a"] = "--testadapterpath",
+                ["-a"] = "--platform",
+                ["-l"] = "--logger",
+                ["-f"] = "--framework",
+                ["-d"] = "--diag",
+                ["--filter"] = "--testcasefilter",
+                ["--list-tests"] = "--listtests",
+                ["--test-adapter-path"] = "--testadapterpath",
+                // .NET 7 breaking change, before we had ["-r"] = "--resultsdirectory",
+                ["--results-directory"] = "--resultsdirectory",
+                ["--arch"] = "--platform"
+            }.ToImmutableDictionary();
+
+        private static readonly ImmutableDictionary<string, string> s_verbosityMapping
+            = new Dictionary<string, string>()
+            {
+                ["q"] = "quiet",
+                ["m"] = "minimal",
+                ["n"] = "normal",
+                ["d"] = "detailed",
+                ["diag"] = "diagnostic"
+            }.ToImmutableDictionary();
+
+        private static readonly ImmutableHashSet<string> s_ignoredArguments
+            = ImmutableHashSet.Create(
+                StringComparer.OrdinalIgnoreCase,
+                "-c",
+                "--configuration",
+                "-r",
+                "--runtime",
+                "-o",
+                "--output",
+                "--no-build",
+                "--no-restore",
+                "--interactive",
+                "--testSessionCorrelationId",
+                "--artifactsProcessingMode-collect"
+            );
+
+        /// <summary>
+        /// The expanded (after applying argument mapping) list of arguments to consider
+        /// as switches (i.e., for which we don't expect a non dashed arg). This also
+        /// includes ignored arguments.
+        /// </summary>
+        private static readonly ImmutableHashSet<string> s_switchArguments
```

Comment:
For all of these arguments, we were always swallowing the dll/exe if it was passed right after.